src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -0,0 +1,511 @@
+        private int _baudRate;
+        private StopBits _stopBits;
+        private Handshake _handshake;
+        private Parity _parity;
+        private int _dataBits = 8;
+        private bool _inBreak = false;
+        private SafeFileHandle _handle = null;
+        private bool _rtsEnable = false;
+        ArrayList selectList;
+        private bool _isAsync = false;
+        private int _readTimeout = 0;
+        private int _writeTimeout = 0;
+
+        // three different events, also wrapped by SerialPort.
+        internal event SerialDataReceivedEventHandler DataReceived;      // called when one character is received.
+        internal event SerialPinChangedEventHandler PinChanged;    // called when any of the pin/ring-related triggers occurs
+        internal event SerialErrorReceivedEventHandler ErrorReceived;         // called when any runtime error occurs on the port (frame, overrun, parity, etc.)
+
+        // ----SECTION: inherited properties from Stream class ------------*
+
+        // These six properties are required for SerialStream to inherit from the abstract Stream class.
+        // Note four of them are always true or false, and two of them throw exceptions, so these
+        // are not usefully queried by applications which know they have a SerialStream, etc...
+
+        public override int ReadTimeout => _readTimeout;
+        public override int WriteTimeout => _writeTimeout;
+
+        public override bool CanRead
+        {
+            get { return (_handle != null); }
+        }
+
+        public override bool CanSeek
+        {
+            get { return false; }
+        }
+
+        public override bool CanTimeout
+        {
+            get { return (_handle != null); }
+        }
+
+        public override bool CanWrite
+        {
+            get { return (_handle != null); }
+        }
+
+        public override long Length
+        {
+            get { throw new NotSupportedException(SR.NotSupported_UnseekableStream); }
+        }
+
+        public override long Position
+        {
+            get { throw new NotSupportedException(SR.NotSupported_UnseekableStream); }
+            set { throw new NotSupportedException(SR.NotSupported_UnseekableStream); }
+        }
+
+        internal int BaudRate
+        {
+            set
+            {
+                if (value != _baudRate)
+                {
+                    if (Interop.Sys.TerminalSetSpeed(_handle, value) != 0)
+                    {
+                        throw new IOException();
+                    }
+                    _baudRate = value;
+                }
+            }
+            get
+            {
+                return Interop.Sys.TerminalGetSpeed(_handle);
+            }
+        }
+
+        public bool BreakState
+        {
+            get { return _inBreak; }
+            set
+            {
+                throw new System.PlatformNotSupportedException(System.SR.PlatformNotSupported_IOPorts);
```

Comment:
nit: remove "System."

Reply:
Do not use this message ("System.IO.Ports is currently only supported on Windows."), create a specific one for this case.